contracts/AutomationBotAggregator.sol

```diff
@@ -160,22 +213,28 @@ contract AutomationBotAggregator {
         uint256[] memory triggerIds
     ) external {
         ManagerLike manager = ManagerLike(serviceRegistry.getRegisteredService(CDP_MANAGER_KEY));
-
-        TriggerGroupRecord memory groupIdRecord = activeTriggerGroups[groupId];
-        bytes32 hash = getTriggerGroupHash(groupId, cdpId, triggerIds);
-        require(hash == groupIdRecord.hash, "aggregator/invalid-trigger-group");
+        require(activeGroups[groupId] == cdpId, "aggregator/inactive-group");
+        for (uint256 i = 0; i < triggerIds.length; i++) {
+            require(groupTriggers[triggerIds[i]] == groupId, "aggregator/inactive-trigger");
+        }
         require(isCdpAllowed(cdpId, msg.sender, manager), "aggregator/no-permissions");
-        activeTriggerGroups[groupId] = TriggerGroupRecord(0, 0);
 
-        emit TriggerGroupRemoved(groupId, triggerIds);
-    }
+        activeGroups[groupId] = 0;
+        for (uint256 i = 0; i < triggerIds.length; i++) {
+            groupTriggers[triggerIds[i]] = 0;
```

Comment:
require that triggerId belongs to the group

Reply:
it's there `require(groupTriggers[triggerIds[i]] == groupId, "aggregator/inactive-trigger");`

Reply:
oh, i see. no need to iterate the array twice then. u can just require before removing

Reply:
don't we waat to check both (in this case) or all triggerIds if they belong to the group ?

Reply:
yeah, but it will revert the transaction if one doesn't match

Reply:
damn me. we were talking about two different things :)